### PR TITLE
arm64: dt: OP-TEE for FVP Base RevC

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Configures FVP Base RevC with OP-TEE.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: arch/arm64/boot/dts/arm/fvp-base-revc.dts [hunk 1]
```diff
@@ -28,7 +28,6 @@
 
 	aliases {
 		serial0 = &v2m_serial0;
-		serial1 = &v2m_serial1;
```
Comment (jforissier) [inline arch/arm64/boot/dts/arm/fvp-base-revc.dts]: Is this intended?

File: arch/arm64/boot/dts/arm/fvp-base-revc.dts [hunk 2]
```diff
@@ -243,4 +247,11 @@
 				<0 0 43 &gic 0 0 GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>,
 				<0 0 44 &gic 0 0 GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>;
 	};
+
+	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
```
Comment (jforissier) [inline arch/arm64/boot/dts/arm/fvp-base-revc.dts:257]: How was the node added to the DT prior to this PR?

**Review by jforissier — COMMENTED** [review]
Could you explain why in the commit description? Thanks.

File: arch/arm64/boot/dts/arm/fvp-base-revc.dts (hunk 1 above)
Comment (jenswikl) [reply arch/arm64/boot/dts/arm/fvp-base-revc.dts]: It was, but I'll restore that for now.

File: arch/arm64/boot/dts/arm/fvp-base-revc.dts (hunk 2 above)
Comment (jenswikl) [reply arch/arm64/boot/dts/arm/fvp-base-revc.dts:257]: It wasn't. We were only using the Foundation model, which needs a different DTB.

Comment (jforissier) [reply arch/arm64/boot/dts/arm/fvp-base-revc.dts:257]: OK, I thought we were using this file with our `fvp.xml` manifest.

Comment (jenswikl) [reply arch/arm64/boot/dts/arm/fvp-base-revc.dts:257]: We are, if compiling with `FVP_USE_BASE_PLAT=y`. It was introduced with commit 1bf41f5d9b0d ("fvp: add Trusted Services support") in https://github.com/OP-TEE/build

**Comment by jenswi-linaro** [comment]
Update

**Comment by jforissier** [comment]
Thanks for the details, all clear now. LGTM.

**Pull request merged**